provider/kv/kv.go

```diff
@@ -114,11 +122,21 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 				}
 			})
 		}
+		if p.CacheKeyValues {
+			tree, err := p.kvClient.List(p.Prefix, nil)
+			if err != nil {
+				return fmt.Errorf("failed to list KV store: %v", err)
+			}
+			kvCache := make(map[string]*store.KVPair, len(tree))
+			for _, kv := range tree {
+				kvCache[kv.Key] = kv
+			}
+			p.kvCache = kvCache
+		}
```

Comment:
Why do we need this L125-L135 block in addition to what's already done in watchKv?